staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -204,6 +206,20 @@ func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) error {
 	c.HealthzChecks = append(c.HealthzChecks, healthz.NamedCheck("etcd", func(r *http.Request) error {
 		return healthCheck()
 	}))
+
+	if s.EncryptionProviderConfigFilepath != "" {
+		healthzConfigs, err := encryptionconfig.GetKMSPluginProbes(s.EncryptionProviderConfigFilepath)
+		if err != nil {
+			return err
+		}
+
+		for i, h := range healthzConfigs {
+			c.HealthzChecks = append(c.HealthzChecks, healthz.NamedCheck(fmt.Sprintf("kms-plugin-%d", i), func(r *http.Request) error {
```

Comment:
Is there something besides the index we can use here? If KMS-plugin has multiple probes and one of them is failing, it may be helpful on the apiserver-side to be able to distinguish the culprit.

Reply:
Yes, added Name filed which uniquely identifies the corresponding KMS Provider.